bin/test_runner.rb

```diff
@@ -17,40 +17,53 @@ class Test
   def initialize(test_in_string_format, jclient, rsclient, opts)
     @jclient = jclient
     @rsclient = rsclient
+    @opts = opts
+    @thresholds = @opts[:thresholds]
+
     elems = test_in_string_format.split(' ')
     @percepts = {}
     if elems.size == 1
       @name = test_in_string_format.chomp
-      update_percepts
-      if opts[:force_run] == false && @percepts[:job_status] == "success"
-        puts "force_run flag not off, and test passed, going straight to Done".fg 'yellow'
+      @cloud_name = cloud_name(@name)
+      if @thresholds.has_key?(@cloud_name) && (@thresholds[@cloud_name] == 0)
+        puts "Cloud #{@cloud_name} has threshold 0, moving to Done"
         @stage = Done
       else
-        puts "New test, set to Staging".fg 'yellow'
-        @stage = Staging
+        init_percepts
+        if opts[:run_anyway] == false && @percepts[:job_status] == "success"
+          puts "run_always flag not off, and test passsed, going straight to Done".fg 'yellow'
+          @stage = Done
+        else
+            puts "New test, set to Staging".fg 'yellow'
+            @stage = Staging
+        end
       end
-    elsif elems.size == 5
+    elsif elems.size == 7
```

Comment:
Top level, if we are only going to check if size of array is either 1 or 7 with an `else` catchall, and this was chagned from 5 to 7, perhaps we should have this be:
```
elsif elems.size > 1
```
so we do not need to update this for future updates.